pkg/cmd/experimental/syncgroups/schema1/memberextractor.go

```diff
@@ -0,0 +1,172 @@
+package schema1
+
+import (
+	"fmt"
+
+	"github.com/go-ldap/ldap"
+
+	ldaputil "github.com/openshift/origin/pkg/auth/authenticator/password/ldappassword/util"
+	"github.com/openshift/origin/pkg/cmd/experimental/syncgroups/config"
+)
+
+// New builds a new GroupMemberExtractor using a schema-appropriate config
+func New(config LDAPSyncConfig) GroupMemberExtractor {
```

Comment:
Need your opinion on `LDAPInterface` for a name - one thing it *can* do is extract members, but it also lists groups, gets a group or member entry for a name, etc.